# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../xpp/xpp-1.5-glibc-2.10.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Fix build agains glibc >= 2.10

Origin: Gentoo Linux

diff -ur xpp-1.5.orig/xpp.cxx xpp-1.5/xpp.cxx
--- xpp-1.5.orig/xpp.cxx	2004-12-06 21:00:04.000000000 +0200
+++ xpp-1.5/xpp.cxx	2009-08-09 17:27:01.000000000 +0300
@@ -168,7 +168,7 @@
 	    {	  
 	      char *tmp;
 
-	      if ((tmp = strchr(dest,'/'))) {
+	      if ((tmp = const_cast <char*> (strchr(dest,'/')))) {
 		tmp[0] = '\0';
 		instance = tmp + 1;
 	      }	
